model/perf_data.go

```diff
@@ -369,102 +369,194 @@ func (ts PerformanceTimeSeries) Statistics() PerformanceStatistics {
 		}
 	}
 
-	return out
+	return out, nil
 }
 
-func (perf *PerformanceStatistics) Mean() (PerformanceMetricSummary, error) {
+func (perf *performanceStatistics) mean() (performanceMetricSummary, error) {
 	var err error
 	catcher := grip.NewBasicCatcher()
-	out := PerformanceMetricSummary{
-		Samples: perf.samples,
-		Span:    perf.span,
+	out := performanceMetricSummary{
+		samples:    perf.samples,
+		span:       perf.span,
+		metricType: "mean",
 	}
-
-	out.TotalTime.Duration = perf.totalTime.duration
-	out.TotalTime.Waiting = perf.totalTime.waiting
-	out.TotalCount.Errors = perf.totalCount.errors
-	out.TotalCount.Size = perf.totalCount.size
-	out.TotalCount.Operations = perf.totalCount.operations
-
-	out.Guages.Failed = perf.state.failed
-
-	out.Counters.Size, err = perf.counters.size.Mean()
+	out.totalTime.duration = perf.totalTime.duration
+	out.totalTime.waiting = perf.totalTime.waiting
+	out.totalCount.errors = perf.totalCount.errors
+	out.totalCount.size = perf.totalCount.size
+	out.totalCount.operations = perf.totalCount.operations
+	out.guages.failed = perf.state.failed
+
+	out.counters.size, err = perf.counters.size.Mean()
 	catcher.Add(err)
 
-	out.Counters.Operations, err = perf.counters.operations.Mean()
+	out.counters.operations, err = perf.counters.operations.Mean()
 	catcher.Add(err)
 
-	out.Counters.Errors, err = perf.counters.errors.Mean()
+	out.counters.errors, err = perf.counters.errors.Mean()
 	catcher.Add(err)
 
-	out.Guages.Workers, err = perf.state.workers.Mean()
+	out.guages.workers, err = perf.state.workers.Mean()
 	catcher.Add(err)
 
-	out.Timers.Duration, err = perf.timers.duration.Mean()
+	out.timers.duration, err = perf.timers.duration.Mean()
 	catcher.Add(err)
 
-	out.Timers.Total, err = perf.timers.total.Mean()
+	out.timers.total, err = perf.timers.total.Mean()
 	catcher.Add(err)
 
 	return out, catcher.Resolve()
 }
 
-func (perf *PerformanceStatistics) P90() (PerformanceMetricSummary, error) {
+func (perf *performanceStatistics) p90() (performanceMetricSummary, error) {
 	return perf.percentile(90.0)
 }
 
-func (perf *PerformanceStatistics) P50() (PerformanceMetricSummary, error) {
+func (perf *performanceStatistics) p50() (performanceMetricSummary, error) {
 	return perf.percentile(50.0)
 }
 
-func (perf *PerformanceStatistics) percentile(pval float64) (PerformanceMetricSummary, error) {
+func (perf *performanceStatistics) percentile(pval float64) (performanceMetricSummary, error) {
 	var err error
 	catcher := grip.NewBasicCatcher()
-	out := PerformanceMetricSummary{
-		Samples: perf.samples,
-		Span:    perf.span,
+	out := performanceMetricSummary{
+		samples:    perf.samples,
+		span:       perf.span,
+		metricType: fmt.Sprintf("percentile_%d", int(pval)),
```

Comment:
you shouldn't need to type cast to int.... use %f (I think.)

Reply:
if I use %f it comes out as like "90.00000" which I think is aggressive personally.

Reply:
You can specify precision with `%f`. You can e.g. `%.2f` if you want 90.00, or `%.f` if you want 90.